src/variational/VariationalInference.jl

```diff
@@ -115,51 +114,57 @@ end
 function grad!(
     vo,
     alg::VariationalInference{<:TrackerAD},
-    q::VariationalPosterior,
-    model::Model,
+    q,
+    model,
     θ::AbstractVector{<:Real},
     out::DiffResults.MutableDiffResult,
     args...
 )
-    θ_tracked = [Tracker.param(θ[i]) for i ∈ eachindex(θ)]
-    y = - vo(alg, q, model, θ_tracked, args...)
+    θ_tracked = Tracker.param(θ)
+    y = if (q isa VariationalPosterior)
```

Comment:
Same here; maybe the function could be reused too?

Reply:
This is resolved right? We're deferring this to AVI